builder/builder.go

```diff
@@ -2015,19 +2015,53 @@ func (b *Builder) BuildDeltaPacksPreviousVersions(prev, to uint32, printReport b
 		cur = m.Header.Previous
 	}
 
-	fmt.Printf("Using %d workers\n", b.NumDeltaWorkers)
 	fmt.Printf("Found %d previous versions\n", len(previousManifests))
 
 	bundleDir := filepath.Join(b.Config.Builder.ServerStateDir, "image")
 	// Create all deltas for all previous versions first based on full manifests
-	for _, fromManifest := range previousManifests {
-		fmt.Println()
-		err = swupd.CreateAllDeltas(outputDir, int(fromManifest.Header.Version), int(toManifest.Header.Version), b.NumDeltaWorkers)
-		if err != nil {
-			return err
-		}
+	var versionQueue = make(chan *swupd.Manifest)
+	var wg sync.WaitGroup
+	var deltaErrors []error
+	versionWorkers := 1
+
+	// If we have at least 2x the number of CPUs as versions, give each version
+	// its own thread to build deltas in.
+	if runtime.NumCPU() >= 2*len(previousManifests) {
```

Comment:
Are we ignoring what the user sets then?

Reply:
Yes in this case until we get better stats on other combinations. It will use their number however for the case that they don't have enough CPUs, which is fine 'cause it won't run multiple version thread pools at least (imo).